schema_based_on_diagram.sql

```diff
@@ -0,0 +1,70 @@
+/* Create Tables */
+
+CREATE TABLE invoice_items (
+  id SERIAL PRIMARY KEY,
+  unit_price DECIMAL,
+  quantity INTEGER,
+  total_price DECIMAL,
+  invoice_id INTEGER,
+  treatment_id INTEGER
+)
+
+
+CREATE TABLE treatments (
+  id SERIAL PRIMARY KEY,
+  type VARCHAR(50),
+  name VARCHAR(100)
+)
+
+CREATE TABLE medical_histories (
+  id SERIAL PRIMARY KEY,
+  admitted_at TIMESTAMP,
+  patient_id INTEGER,
+  status VARCHAR(50)
+)
+
+CREATE TABLE patients (
+  id SERIAL PRIMARY KEY,
+  name VARCHAR(100),
+  date_of_birth DATE
+)
+
+CREATE TABLE invoices (
+  id SERIAL PRIMARY KEY,
+  total_amount DECIMAL,
+  generated_at TIMESTAMP,
+  payed_at TIMESTAMP,
+  medical_history_id INTEGER
+)
+
+/* Set Foreign Keys */
+
+/* medical_histories */
+
+BEGIN;
+
+ALTER TABLE medical_histories
+ADD FOREIGN KEY (patient_id) REFERENCES patients(id);
+
+ALTER TABLE medical_histories
+ADD FOREIGN KEY (id) REFERENCES treatments(id);
+
```

Comment:
- [ ] Please remove the foreign key constraint from the `id` column as this column is considered as a primary key for the `medical_histories` table.